### PR TITLE
Add React Native (iOS and Android) support

### PR DESCRIPTION
**Pull request by acostalima**
### TODO

- [ ] Add types for `react-native-fetch-api`
- [ ] Integrate `rn-test` with `aegir` to run tests for both iOS and Android
- [ ] Finish CI and setup and ensure it runs to completion successfully

**NOTE:** React Native does not support `ReadableStream` for request bodies. However, just like the browser, we can use `XMLHttpRequest.upload.onprogress` event as a workaround.

@hugomrdias, can you give me a hand on items above? 🙏 

Depends on https://github.com/hugomrdias/iso-url/pull/154

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -68,7 +76,12 @@
     "it-all": "^1.0.4",
     "it-drain": "^1.0.3",
     "it-last": "^1.0.4",
-    "uint8arrays": "^2.0.5"
+    "react-native-polyfill-globals": "^3.0.0",
+    "react-native-test-runner": "^2.3.0",
+    "react-native-url-polyfill": "^1.2.0",
+    "text-encoding": "^0.7.0",
+    "uint8arrays": "^2.0.5",
+    "web-streams-polyfill": "^3.0.1"
```
Comment (hugomrdias) [inline package.json:84]: why do we need this here ?

**Review by hugomrdias — CHANGES_REQUESTED** [review]

File: src/text-encoder.js [hunk 2]
```diff
@@ -1,5 +1,4 @@
 'use strict'
 
-const { TextEncoder } = require('web-encoding')
-
-module.exports = TextEncoder
+// This is fine for React Native as long as the environment is polyfilled with text-encoding
+module.exports = require('web-encoding').TextEncoder
```
Comment (hugomrdias) [inline src/text-encoder.js:4]: can you check this PR https://github.com/Gozala/web-encoding/pull/13 and see if its enough ?

**Review by hugomrdias — CHANGES_REQUESTED** [review]

File: src/text-encoder.js (hunk 2 above)
Comment (acostalima) [reply src/text-encoder.js:4]: Yep, it should work. I'll have a look. Thanks!

File: package.json (hunk 1 above)
Comment (acostalima) [reply package.json:84]: We should be able to move this to RN test runner's config so it is installed in the test app directly instead. The same goes for `react-native-url-polyfill`.

Comment (acostalima) [reply package.json:84]: Sorry, I misunderstood your comment. 🙏 
Yes, we don't need web streams here but only in HTTP clien't test suite.

Comment (hugomrdias) [reply package.json:84]: maybe they can live in the `react-native-polyfill-globals` package

Comment (acostalima) [reply package.json:84]: I disagree. Users of `react-native-polyfill-globals` might want to polyfill only a few APIs. If all these dependencies live there, everything is pulled once the package is installed. If we don't want to pollute `js-ipfs-utils` dependency list with such dependencies, we can install them in the test app via its config.

File: src/text-encoder.js (hunk 2 above)
Comment (acostalima) [reply src/text-encoder.js:4]: It seems to be working fine! I'm updating `web-encoding` to the latest version.

File: package.json (hunk 1 above)
Comment (acostalima) [reply package.json]: `web-streams-polyfill` and `react-native-url-polyfill` are now being installed in the test app.

File: src/text-encoder.js (hunk 2 above)
Comment (acostalima) [reply src/text-encoder.js:3]: I've just removed the comment because it is no longer necessary.

**Comment by acostalima** [comment]
Without `aegir` integration, tests can be run with the following commands:

```
$ node .aegir.js & # with the temporary changes introduced to start the echo server without aegir
$ AEGIR_RUNNER=react-native ECHO_SERVER=http://localhost:3000 npx rn-test --platform ios --simulator 'iPhone 11 (14.1)' 'test/**/*.spec.js'
```

**Pull request merged**